.github/workflows/ci.yaml

```diff
@@ -18,44 +18,131 @@ jobs:
       run: echo ${{ github.ref }}
     - name: event_name
       run: echo ${{ github.event_name }}
-  build:
-    name: Build
+  lint:
+    name: Lint
+    runs-on: ubuntu-latest
+    steps:
+    - name: checkout
+      uses: actions/checkout@v3
+    - uses: actions/setup-go@v2
+      with:
+        go-version: 1.17
+    - name: golangci-lint
+      uses: golangci/golangci-lint-action@v3.1.0
+      with:
+        version: v1.44.2
```

Comment:
Separate lint job to help speed up runs, the golangci-lint-action also provides additional caching and optimization beyond running it through make.